ABM/Modules/Demographic_functions.jl

```diff
@@ -0,0 +1,765 @@
+    #% WITHIN PATCH LONG DISTANCE DISPERSAL
+    """ 
+    # Within patch long distance dispersal
+    Disperse a fraction of seeds beyond the parent crown but within the same habitat fragment. The
+    disperesed seeds are added to the seedlings bank for the appropriate grid cell.
+    ## Arguments:
+    - `model`: The AgentBasedModel object defining the current model. This object is usually 
+    created using `Agents.ABM()`.
+    - `seed_production::Int64`: Species-specific constant value used within a Poisson distribution
+    to determine the number of seeds produced by an agent.
+    - `ldd_disp_frac::Float64`: Proportion of seedlings that undergo long-distance dispersal and are
+    therefore unavailable for neighbourhood dispersal.
+    - `ldd_dispersal_dist::Int64`: Species-specific mean dispersal distance
+    - `cell_grain::Int64`: The size of an individual cell in the model (cell_grain x cell_grain).
+    - `pos::Tuple{Int64,Int64}`: Coordinates of tree/agent.
+    - `pcors::Vector{Vector{Tuple{Int64, Int64}}}`: List of coordinates for all cells in model grid.
+    - `shad_h::Vector{Float64}`: Mean shade heights as calculated by `get_nhb_shade_height()` for 
+    all cells in model grid.
+    - `r_hgt::Int64`: Regeneration height in meters for selected species.
+    - `seedlings::Vector{Vector{Int64}}`: Number of seedlings for each species for every cell in the
+    model grid.
+    - `spec_ID::Int64`: Selected species ID.
+    """
+    function ldd_within(
+        model,
+        seed_production::Int64,
+        ldd_disp_frac::Float64,
+        ldd_dispersal_dist::Int64,
+        cell_grain::Int64,
+        pos::Tuple{Int64,Int64},
+        pcors::Vector{Vector{Tuple{Int64, Int64}}},
+        shad_h::Vector{Float64},
+        r_hgt::Int64,
+        seedlings::Vector{Vector{Int64}},
+        spec_ID::Int64
+    )
+        #% GENERATE SEEDS TO BE DISPERSED---------------------------#
+        ldd_seeds = trunc(Int, (rand(Poisson(seed_production)) 
+                                            .* (ldd_disp_frac)))
+
+        #% LOOP THROUGH ALL SEEDS AND ASSIGN TO A CELL--------------#
+        #Threads.@threads for _ in 1:ldd_seeds
+        for _ in 1:ldd_seeds
+            #Calculate appropriate distance
+            D = trunc(Int, rand(Exponential((ldd_dispersal_dist) ./ cell_grain)));
+            
+            #TODO Move this documentation to the top of the function
+            #* Find cells at distance D only. To do this currently we find all the
+            #* cells within distance D and remove the ones also within distance D-1
+            D_nhbs = set_get_functions.positions_at(pos::Tuple{Int64,Int64}, 
+                                                    model::ABM{<:GridSpaceSingle}, 
+                                                    D::Int64,)
+
+            ## If there are avaible cells select one to be the target
+            target = Int64[]
+            if isempty(D_nhbs) .== false
```

Comment:
Not sure I understand the broadcasting here (the use of dot in `.==`); `isempty` returns `true` or `false` not a vector here. Perhaps you mean:

```suggestion
            if !isempty(D_nhbs)
```
?